src/environmentd/src/deployment/preflight.rs

```diff
@@ -109,13 +109,18 @@ pub async fn preflight_legacy(
         tracing::info!("Waiting for user to promote this envd to leader");
         promoted.await;
 
-        Ok(mz_catalog::durable::persist_backed_catalog_state(
+        let catalog = mz_catalog::durable::persist_backed_catalog_state(
             persist_client,
             environment_id.organization_id(),
             BUILD_INFO.semver_version(),
             Arc::clone(&catalog_metrics),
         )
-        .await?)
+        .await?;
+
+        // Once we've opened the catalog, we've become the leader.
```

Comment:
I think this comment is a little misleading. We have opened the catalog in savepoint mode, but we have to re-open it in the normal writable mode. That is usually not a very fast process.

Reply:
Oh, I think this is a conceptual error, now that you mention it. This is just opening the adapter storage, which does _not_ do fencing.

Reply:
We actually haven't even opened the adapter storage. As you'll note, `catalog` is of type `OpenableDurableCatalogState`. First we have to open the adapter storage (durable catalog) via `let catalog = catalog.open(...)`, then we have to open in memory catalog via `Catalog::open(catalog, ...)`.

Reply:
The first open call is what executes the fencing.

Reply:
Yeah, can you take a look at the latest revision? I just adjusted it to move the call to `set_is_leader` after we open the openable durable catalog state.

Reply:
And I'm pretty sure that we don't need to open the catalog entirely, but just the openable durable catalog state, because it's the openable durable catalog state that handles the deploy generation fencing AFAICT.

Reply:
>  it's the openable durable catalog state that handles the deploy generation fencing AFAICT.

That's correct.